gtars/src/uniwig/writing.rs

```diff
@@ -0,0 +1,117 @@
+use crate::uniwig::Chromosome;
+use ndarray::Array;
+use ndarray_npy::write_npy;
+use std::fs::{create_dir_all, remove_file, File, OpenOptions};
+use std::io;
+use std::io::{BufWriter, Write};
+
+pub fn write_to_npy_file(
+    counts: &Vec<u32>,
+    filename: String,
+    chromname: String,
+    start_position: i32,
+    stepsize: i32,
+    metafilename: String,
+) {
+    // For future reference `&Vec<u32>` is a SLICE and thus we must use the `to_vec` function below when creating an array
+    // https://users.rust-lang.org/t/why-does-std-to-vec-exist/45893/9
+
+    // Write the NumPy Files
+    let arr = Array::from_vec(counts.to_vec());
+    let _ = write_npy(filename, &arr);
+
+    // Write to the metadata file.
+    // Note: there should be a single metadata file for starts, ends and core
+
+    let path = std::path::Path::new(&metafilename).parent().unwrap();
+    let _ = create_dir_all(path);
+
+    let mut file = OpenOptions::new()
+        .create(true) // Create the file if it doesn't exist
+        .append(true) // Append data to the existing file if it does exist
+        .open(metafilename)
+        .unwrap();
+
+    // The original wiggle file header. This can be anything we wish it to be. Currently space delimited.
+    let mut wig_header = "fixedStep chrom=".to_string()
+        + chromname.as_str()
+        + " start="
+        + start_position.to_string().as_str()
+        + " step="
+        + stepsize.to_string().as_str();
+    wig_header.push_str("\n");
```

Comment:
Since `\n` is a `char`. It might be better to change to `push('\n')` (note the single quotes

Reply:
Yeah. That worked!